template/registry/registry.go

```diff
@@ -0,0 +1,146 @@
+package registry
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Registry map[string]Func
+
+// Func is a function that can be used in formatting and matching contexts.
+type Func struct {
+	FormatSignature *Signature                             // Signature of the function when called in formatting context
+	MatchSignature  *Signature                             // Signature of the function when called in matching context
+	Fn              func(any, map[string]any) (any, error) // Function itself
```

Comment:
Is there a case when first returned value is other than string?

Reply:
it could be anything :), thats the problem.